### DISCUSSION TITLE
Imported Script from /lib triggers first

### DISCUSSION THREAD

**Discussion by zacjones93** [johnlindquist/kit#273] [Q&A] [answered]
The main script I'm building takes in an egghead course slug, queries for the data from the API, and after some manipulation of the data it gets posts an object to Sanity. This script works great.

Part of the problem I want to solve is to add additional information to that object that gets posted to Sanity. In addition to egghead course data, we have illustrators (in Sanity) that I can loop through and display to the user to select given they know who illustrated the art for a particular course.

It seemed like the right idea to create this additional feature in a `lib` folder and import it into my original script rather than bloat the original script.

The **problem** I run into is that the exported `arg` call is called immediately where I would prefer that I could control when it gets called. 

https://www.loom.com/share/a487c55ac7234999b263c34cd13115b8

<details>
<summary>lib/sanity.js</summary>

<br>

```js
const sanityClient = await npm("@sanity/client");
let { GraphQLClient, gql } = await npm("graphql-request");
let { nanoid } = await npm("nanoid");
let groq = await npm("groq");

let key = await env("SANITY_READ_WRITE_KEY");

const client = sanityClient({
...
});

let illustratorQuery = groq`*[_type == 'collaborator' && role == 'illustrator'][]{
  'name': person->name,
  'slug': person->slug.current,
  'image': person->image.url
}`;

let illustrators = await client.fetch(illustratorQuery);

illustrators.push({
  name: "None",
  value: null,
  image: "",
});

export let selectedIllustrator = await arg("Select an Illustrator", (input) => {
  return illustrators.map((illustrator) => {
    return {
      name: illustrator.name,
      value: illustrator.slug,
      img: illustrator.image,
    };
  });
});
```

</details>

**Comment by johnlindquist** [comment, marked as answer]
Cool script 😎

"libs" imply exporting functions without any "top-level" code.

As soon as you import a lib, all the code runs. So something like:

```js
export let selectedIllustrator = await arg("Select an Illustrator", (input) => {
  return illustrators.map((illustrator) => {
    return {
      name: illustrator.name,
      value: illustrator.slug,
      img: illustrator.image,
    };
  });
});
```

Should become:
```js
export let selectedIllustrator = async ()=> await arg("Select an Illustrator", (input) => {
  return illustrators.map((illustrator) => {
    return {
      name: illustrator.name,
      value: illustrator.slug,
      img: illustrator.image,
    };
  });
});
```

Top-level `await` is _great_ for scripts, but quickly becomes this exact problem for "apps".

**Reply by zacjones93** [reply]
Ahhh ok, that makes sense! I had a feeling wrapping the function was the way to go but wasn't quite sure how to approach it, this is real simple.

And it works!
